HIP/hip-21.md

```diff
@@ -32,41 +33,70 @@ Today, applications have to use the File Service and submit two file contents qu
 
 ## Rationale 
 
-The proposal is to create a free non-file HAPI query to return network information. As we continue to grow the network and transition into a permissionless model, the number of times the local address book will be required to update will increase. The transaction fee associated with the current file query hinders the ability to update the address book as frequently as it might be needed for applications to ensure their address book is always up to date. This information should be free. 
+The proposal is to create a rest API query to return network information. As we continue to grow the network and transition into a permissionless model, the number of times the local address book will be required to update will increase. The transaction fee associated with the current file query hinders the ability to update the address book as frequently as it might be needed for applications to ensure their address book is always up to date. This information should be free. 
 
 ## Specification 
 
-- Add a non-file HAPI query called **getNetworkInfo** to the existing Network Service
-- This query would not have a transaction fee
+REST API
+
+GET `api/v1/addressbooks`
+
+Response:
+
+```json
+
+{
+  "addressbook": [
```

Comment:
Here is the suggested syntax with that matches the mirror node database schema and using actual data from mainnet:
```json
{
  "nodes": [
    {
      "description": "",
      "memo": "0.0.3",
      "node_id": 0,
      "node_account_id": "0.0.3",
      "node_cert_hash": "0x333432343464353061386564346434636261646632353632306431616238386133323038313937376432663864373062613832626135326233333035613435363038323463663662303130366436333565643563333932373266666661626538",
      "public_key": "0x308201a2300d06092a864886f70d01010105000382018f003082018a02820181009098865def2f2ab376c7f0f738c1d87a27ac01afd008620c35cb6ebfcbb0c33003193a388c346d30231727012193bb76fd3004b864312c689ef5213cbb901101509deab94f26a732e637929da4c4cb32517e3adbb3811d50ac4c77c1fce8b651606215f34707f3e7265545e58c894609e28376bdb7775fe30439e0e1592fdcb0c3ee1c305773d072a6b8957eafce1a11be965edaff3843366cb6a44ec25a890106e6247567f76b550fda482baec6307d698ec88841fd66f23f210e47b8a9dcba6ba4e1fa716db33c80e30819496dcb5e5609fb6e7c615379bdded427e9231b9254c2baf943608a86d698ae9a3c8639df887d6f6b5a71385d24338d911a212bf71f1e2acc8b186b96ec8e69c86b6d058217776a09c9c68953edb5916578b5a263b2f469e3b0c07eada71a447eea7f8fc1bb8074255567b7f0bd1e6afb0358718c98b429e24b2298596fc76cf6af396ca9434d7926ec7d37d4b92af56d45feff8196095224a916c1ffe6b667e255fc3ac8cccef920dc044b25003132b87806742f0203010001",
      "service_endpoints": [
        {
          "ip_address_v4": "13.124.142.126",
          "port": 50211
        },
        {
          "ip_address_v4": "13.124.142.126",
          "port": 50212
        },
        {
          "ip_address_v4": "13.82.40.153",
          "port": 50211
        }
      ],
      "stake": 0,
      "timestamp": "1636052707.740848001"
    }
  ],
  "links": {
    "next": null
  }
}
```